### PR TITLE
docs: Introduce an user guide

### PR DESCRIPTION
**Pull request by traxys**
This user guide currently only documents how to install & some frequently asked questions. It also includes the previously introduced documentation on helpers.

### CODE REVIEW

File: docs/user-guide/faq.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+extraPlugins = [(pkgs.vimUtils.buildVimPlugin {
+    name = "my-plugin";
+    src = pkgs.fetchFromGitHub {
+        owner = "<owner>";
+        repo = "<repo>";
+        rev = "<commit hash>";
+        hash = "<nix NAR hash>";
+    };
+})];
+```
```
Comment (GaetanLepage) [inline docs/user-guide/faq.md:24]:
```suggestion
extraPlugins = [
  (pkgs.vimUtils.buildVimPlugin {
    name = "my-plugin";
    src = pkgs.fetchFromGitHub {
        owner = "<owner>";
        repo = "<repo>";
        rev = "<commit hash>";
        hash = "<nix NAR hash>";
    };
  })
];
```
```

**Comment by GaetanLepage** [comment]
That is a very good idea !

File: docs/user-guide/faq.md (hunk 1 above)
Comment (traxys) [reply docs/user-guide/faq.md:24]: I specifically don't want to do that, in order to reduce the length of the example section.

Comment (GaetanLepage) [reply docs/user-guide/faq.md:24]: Ok, as you want.
Personally, I find it more readable to break a line but you get to choose :)

Comment (traxys) [reply docs/user-guide/faq.md:24]: I agree on readability of the code, but reducing the length of the doc will increase overall readability I think

**Review by GaetanLepage — APPROVED** [review]
LGTM ! Great work :)

**Pull request merged**